src/ziggurat/streams.clj

```diff
@@ -173,20 +243,39 @@
     (.build builder)))
 
 (defn- start-stream* [handler-fn stream-config topic-entity channels]
-  (KafkaStreams. ^Topology (topology handler-fn stream-config topic-entity channels)
-                 ^Properties (properties stream-config)
-                 (new TracingKafkaClientSupplier tracer)))
+  (let [topology-fn (case (:consumer-type stream-config)
+                      :stream-joins stream-joins-topology
+                      topology)
+        top         (topology-fn handler-fn stream-config topic-entity channels)
+        _           (log/info (.describe top))]
+    (KafkaStreams. ^Topology top
+                   ^Properties (properties stream-config)
+                   (new TracingKafkaClientSupplier tracer))))
+
+(defn- merge-stream-joins-config ;; not sure about this, :consumer-type can be from the config.edn or the map from init-fn, please comment!
```

Comment:
let's use consumer-type from config. My reason for this is to keep it consistent with other configs.

Reply:
done